Upversion and style cleanup

TheDuckCow opened the pull request:
This should be the last PR for v3.6.2, just doing some final QA testing.

Edit: Uncovered lots of other issues with 5.0 and some gaps in the unit test coverage. So, this PR is now addressing all of that.

TheDuckCow committed:
Some style cleanup and alt form of the exception bounds loop

TheDuckCow committed:
Version and pot update

TheDuckCow committed:
Support color split node across more versions

For some reason, they changed the colorsplit node in 3.3, then changed
it back to how it was in 3.2 and earlier with 5.0.

StandingPadAnimations commented on MCprep_addon/materials/generate.py:
```diff
@@ -1121,8 +1121,16 @@ def texgen_seus(mat: Material, passes: Dict[str, Image], nodeInputs: List, use_r
 		location=(-80, -280))
+	sep_node = ""
+	if util.min_bv((5, 0)):
+		sep_node = "ShaderNodeSeparateRGB"
+	elif util.min_bv((3, 3)):
+		sep_node = "ShaderNodeSeparateColor"
+	else:
+		sep_node = "ShaderNodeSeparateRGB"
```
Slight nitpick: since 3.3-5.0 is the only exception so far, I think it would make sense to simplify this down:


```suggestion
	sep_node = "ShaderNodeSeparateRGB"
	if util.min_bv((3, 3)) and not util.min_bv((5, 0)):
		sep_node = "ShaderNodeSeparateColor"
```

TheDuckCow replied:
Sure, I also realize that I got it mixed up partially since by tests weren't covering 5.0 like I thought, there's only one conditional now actually, not as crazy as I thought it was initially.

StandingPadAnimations commented on MCprep_addon/materials/generate.py:
```diff
@@ -1145,9 +1153,9 @@ def texgen_seus(mat: Material, passes: Dict[str, Image], nodeInputs: List, use_r
-	GREEN = "G" if util.min_bv((5, 0)) else "Green"
+	RED = "Red" if (3, 3) < bpy.app.version < (5, 0) else "R"
+	BLUE = "Blue" if (3, 3) < bpy.app.version < (5, 0) else "B"
+	GREEN = "Green" if (3, 3) < bpy.app.version < (5, 0) else "G"
```
Might be a good time to create some sort of `in_bv_range` function in `util.py`, given both the above conditional and this section. Something like this:

```py
def in_bv_range(lower_bound: Tuple, upper_bound: Tuple, inclusive: bool = True) -> bool:
    return min_bv(lower_bound, inclusive) and not min_bv(upper_bound, inclusive)
```

TheDuckCow replied:
Per above, only one conditional now thankfully. Longer term, I'm seeing less of a need for this these bv utils as we no longer support any Blender version that doesn't yet have the bpy.app.version field (which is why the abstraction function was added anyhow). They don't save that much space and are awkward to place a tuple in (double nesting). The bv4 and bv5 ones I can see an argument for though since they are shorter.

StandingPadAnimations commented on MCprep_addon/__init__.py:
```diff
@@ -41,7 +41,7 @@
 	"category": "Object",
-	"version": (3, 6, 1, 3),
+	"version": (3, 6, 2),
 	"blender": (2, 80, 0),
```
Should probably update this to 2.83, since we no longer include 2.8-2.82 in the test suite or the BpyBuild config

StandingPadAnimations replied:
Speaking of this, we should also update the `_blender_manifest.toml` file. While we don't provide extension builds currently, we do have the configuration (minus `build_extension` being set to `false`) and actions needed to build an extension build, so we also need to update the `version` property in the manifest

https://github.com/Moo-Ack-Productions/MCprep/blob/7fdcf6f59c5129e7bda45d9cdbce78a4312a2b44/MCprep_addon/_blender_manifest.toml#L4

TheDuckCow committed:
Actually correcting the split node colors across versions

TheDuckCow committed:
Updated blender 5.0 for slots

TheDuckCow committed:
Added additional tests to help validate fixes upcoming for Blender 5.0

TheDuckCow committed:
Fix UV scaling in Blender 5.0 to not do redundant UV selection check

These functions to check UV loop selections were removed, but it's not
necessary anyways since Blender 5.0 mandates that face selection ==
UV face selection in the 3D view and UV editor view at once, and we
already do the polygon face selection check earlier on.

TheDuckCow committed:
Updated data mappings for 1.21.10 latest and updated meshswap removables
